### ISSUE TITLE
Panic when clicking on 'Challenge' -> 'Traffic signal survivor'

### ISSUE THREAD

**Issue by strct** [a-b-street/abstreet#515]
Clicking on 'Challenge' -> 'Traffic signal survivor' always crashes my abstreet.

OS: Debian 10
Version: [v0.2.30](https://github.com/a-b-street/abstreet/releases/tag/v0.2.30)

```gdb
SpawnOverTime each agent (10)... 0.0000s

BorderSpawnOverTime (41)... 0.0002s
Generating scenario small_run... plus 0.0001s
Generating scenario small_run took 0.0004s
[2021-02-14T16:11:43Z WARN  sim::make::scenario] 0 of 1,740 people have nonsense schedules
Instantiating small_run...

trips for People (1,740)... 0.0006s

seed parked cars (247)... 0.0003s

spawn trips (1,740)... 0.0010s
Instantiating small_run... plus 0.0010s
Instantiating small_run took 0.0029s
screensaver traffic... plus 0.0091s
screensaver traffic took 0.0124s

- screensaver traffic took 0.0124s
  - Generating scenario small_run took 0.0004s
    - SpawnOverTime each agent (100)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - SpawnOverTime each agent (10)... 0.0000s
    - BorderSpawnOverTime (41)... 0.0002s
    - ... plus 0.0001s
  - Instantiating small_run took 0.0029s
    - trips for People (1,740)... 0.0006s
    - seed parked cars (247)... 0.0003s
    - spawn trips (1,740)... 0.0010s
    - ... plus 0.0010s
  - ... plus 0.0091s
load ../data/system/seattle/maps/montlake.bin... plus 0.0196s
load ../data/system/seattle/maps/montlake.bin took 0.1838s

- load ../data/system/seattle/maps/montlake.bin took 0.1838s
  - Read ../data/system/seattle/maps/montlake.bin (3)... 0.0545s
  - draw_map took 0.1097s
    - make DrawRoads (590)... 0.0001s
    - make DrawLanes (2,848)... 0.0053s
    - make DrawIntersections (445)... 0.0001s
    - generate unzoomed roads and intersections took 0.0020s
    - make DrawBuildings (3,080)... 0.0324s
    - upload all buildings took 0.0099s
    - make DrawParkingLot took 0.0012s
    - make DrawAreas (67)... 0.0000s
    - upload all areas took 0.0000s
    - create quadtree took 0.0584s
    - ... plus 0.0002s
  - ... plus 0.0196s
[2021-02-14T16:11:45Z WARN  usvg::svgtree::parse] Failed to parse fill value: ''. Fallback to black.
thread 'main' panicked at 'Two buttons in one Panel both use action Traffic signal survivor', widgetry/src/widgets/mod.rs:646:17
stack backtrace:
   0: rust_begin_unwind
             at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/panicking.rs:475
   1: std::panicking::begin_panic_fmt
             at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/panicking.rs:429
   2: widgetry::widgets::Widget::get_all_click_actions
   3: widgetry::widgets::Widget::get_all_click_actions
   4: widgetry::widgets::Widget::get_all_click_actions
   5: widgetry::widgets::Widget::get_all_click_actions
   6: widgetry::widgets::panel::PanelBuilder::build_custom
   7: game::challenges::ChallengesPicker::make
   8: <game::challenges::ChallengesPicker as widgetry::app_state::State<game::app::App>>::event
   9: widgetry::app_state::App<A>::event
  10: <std::panic::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
  11: widgetry::runner::State<A>::event
  12: widgetry::runner::run::{{closure}}
  13: winit::platform_impl::platform::sticky_exit_callback
  14: winit::platform_impl::platform::x11::event_processor::EventProcessor<T>::process_event
  15: winit::platform_impl::platform::x11::EventLoop<T>::run
  16: winit::platform_impl::platform::EventLoop<T>::run
  17: winit::event_loop::EventLoop<T>::run
  18: widgetry::runner::run
  19: game::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

********************************************************************************
Wrote ../data/player/camera_state/seattle/montlake.json
Crash! Please report to https://github.com/a-b-street/abstreet/issues/ and include all output.txt; at least everything starting from the stack trace above!

At 00:00:00.0

Camera:
{ "cam_x": 12821.666000000001, "cam_y": 14453.667, "cam_zoom": 10 }

No edits

Crash! Please report to https://github.com/a-b-street/abstreet/issues/ and include all output.txt; at least everything above here until the start of the report!
********************************************************************************
```

**Comment by dabreegster** [comment]
Thanks for reporting! Fixed in 0f1c30a889bbcf1d3619bee388168b42da6eba45, it'll be in the next release in 1-2 hours

**Comment by dabreegster** [comment]
Fixed in https://github.com/a-b-street/abstreet/releases/tag/v0.2.31. Let me know if you hit more problems!
